### PR TITLE
Set json library on OAuth2

### PR DESCRIPTION
**Pull request by snewcomer**
Based on https://github.com/scrogson/oauth2/pull/92/files

https://github.com/ueberauth/ueberauth_twitter/pull/27
https://github.com/ueberauth/ueberauth_twitter/pull/28
https://github.com/ueberauth/ueberauth/pull/88

### CODE REVIEW

**Comment by yordis** [comment]
@snewcomer Could you use https://github.com/ueberauth/ueberauth/blob/2087e031c3ff486e0aafb4684659a3e224d2aa29/lib/ueberauth.ex#L190 instead?

**Comment by mspanc** [comment]
Hi, any chance to release this soon? I can help if necessary.

Currently if you use the most recent versions of ueberauth and ueberauth_google they do not work well togehter.

ueberauth will try to use Jason

```
iex(2)> Ueberauth.json_library
Jason
```

but ueberauth_google will try to use Poison

ueberauth 0.6.1
ueberauth_google 88b9ae851ffda0d34e991ee85e84a0803f7448d0

**Comment by snewcomer** [comment]
Ah I'll update right now!

**Comment by snewcomer** [comment]
@mspanc Updated.  I was searching around for quite a while since I knew there was another PR somewhere.  Thx for helping dig it out!

**Comment by mspanc** [comment]
@yordis any chance for merging/releasing?

**Comment by yordis** [comment]
@mspanc I am confused, the PR does not use `Ueberauth.json_library` unless I am missing something

File: lib/ueberauth/strategy/google/oauth.ex [hunk 1]
```diff
@@ -27,7 +27,10 @@ defmodule Ueberauth.Strategy.Google.OAuth do
   def client(opts \\ []) do
     config = Application.get_env(:ueberauth, __MODULE__, [])
     opts = @defaults |> Keyword.merge(opts) |> Keyword.merge(config) |> resolve_values()
+    json_library = Application.get_env(:ueberauth, :json_library)
```
Comment (mspanc) [inline lib/ueberauth/strategy/google/oauth.ex]: @snewcomer 

shouldn't that be 

```elixir
json_library = Ueberauth.json_library()
```

?

Comment (snewcomer) [reply lib/ueberauth/strategy/google/oauth.ex]: glad somebody looked at my code 😆

Comment (yordis) [reply lib/ueberauth/strategy/google/oauth.ex]: @snewcomer 😆

**Comment by yordis** [comment]
@snewcomer 🚀

**Comment by doomspork** [comment]
@yordis are we ready to cut a new release? If so, will you prep the PR to do so?

**Comment by yordis** [comment]
@doomspork I want to start adding all the for GitHub like such https://github.com/straw-hat-team/straw_hat/tree/master/.github

I just need to add the checkbox in Pull Request asking people to add a log to the `CHANGELOG` file (I need to update Straw Hat as well)

I will be working on this today for sure.

**Comment by rschooley** [comment]
Could we get a release with this update?  v0.8.0 breaks on the hard poison dependency.
Thanks for fixing it.

**Pull request merged**